Release/1.0.1

oguzhanunlu opened the pull request.

pondzix commented on .github/workflows/ci.yml:
```diff
@@ -0,0 +1,36 @@
+on:
+  push:
+    tags:
+      - '*'
```
Do we need to run this flow on tag push? Isn't tag only for 'release' stuff?

pondzix commented on .github/workflows/release.yml:
```diff
@@ -0,0 +1,21 @@
+      - '*'
+
+jobs:
+  release:
```
I can see we have `gh-pages` branch but no step for publishing docs. We could do similar thing as [here](https://github.com/snowplow-incubator/scala-lru-map/blob/e60cd3b63191fe7f2c40b2da6552d1b04664b07d/.github/workflows/release.yml#L15), so use `sbt makeSite` and then use action to deploy to branch. Related [issue](https://github.com/snowplow-incubator/scala-lru-map/issues/53)

pondzix commented on project/BuildSettings.scala:
```diff
@@ -26,33 +23,30 @@ import com.typesafe.sbt.SbtGit.GitKeys._
   lazy val buildSettings = Seq[Setting[_]](
     organization := "com.snowplowanalytics",
-    scalaVersion := "2.12.8"
+    scalaVersion := "2.12.15",
```
What do you think about bump to 2.13.8 and cross compile to 2.12.15?

pondzix commented on project/Dependencies.scala:
```diff
@@ -21,10 +21,9 @@ object Dependencies {
-    val circe      = "0.13.0"
-    val lruMap     = "0.5.0"
-    val scalaj     = "2.4.1"
+    val circe  = "0.13.0"
```
bump circe -> 0.14.1?

pondzix commented on project/Dependencies.scala:
```diff
@@ -36,7 +35,6 @@ object Dependencies {
     val jodaConvert = "org.joda" % "joda-convert" % V.jodaConvert
 
     // Scala
-    val catsEffect  = "org.typelevel"         %% "cats-effect"   % V.catsEffect
```
I can see we use cats-effect directly in the code (`Sync`) , so I think it should still be here defined as dependency.

oguzhanunlu replied:
what's the point of having transitive dep versions? we also use other transitive libs in the codebase but don't maintain their versions separately. That would also get evicted when a future maintainer forgets to bump lru map without bumping cats-effect (in a new lru map release bumping cats-effect version)

pondzix replied:
I think defining all direct dependencies and not relying on transitive ones can make life of maintainers easier. Often, when I don't know some lib I just look at direct dependencies and can quickly assume what to expect more or less like `ok, this lib uses some cats-effect stuff, upgrade may not be trivial` 🙂 Then I go to the code, see some `cats-effect` things and surprise - I have no idea where does it come from and what version it is. 

When you upgrade/replace/remove some direct dependency (e.g. responsible for some private internal things, not exposed in our library API) and it turns out half of our code doesn't compile, because some transitive dependencies are also gone, it can be surprising and take more time to fix. IMO the more things we do explicitly, the less unexpected and confusing problems we going to have.

If it's evicted (since sbt [1.5.0](https://github.com/sbt/sbt/releases/tag/v1.5.0) there is an eviction error based on used versioning) - good, it means that there some conflict which needs to be fixed, the sooner the better 🙂 I'm not saying we should now inspect whole our code base and detect all direct dependencies and add them to our build, but we could at least preserve the ones we have now.

oguzhanunlu replied:
I guess it boils down to personal preference and I see your points. Let me share how I tackle cases like that.

> I have no idea where does it come from and what version it is.

Since [sbt 1.4](https://www.scala-sbt.org/1.x/docs/sbt-1.4-Release-Notes.html#sbt-dependency-graph+is+in-sourced), `sbt-dependency-graph` plugin is in-sourced, which brings tasks like `dependencyBrowseTree`, `whatDependsOn` etc.

For small projects, I prefer `dependencyTree` since it fits into terminal. For big projects, I use either `dependencyBrowseTree` or `whatDependsOn`

```
scala-weather> dependencyTree
[info] com.snowplowanalytics:scala-weather_2.13:1.0.0+9-370dff64+20220301-2048-SNAPSH..
[info]   +-com.snowplowanalytics:scala-lru-map_2.13:0.6.0 [S]
[info]   | +-com.github.blemale:scaffeine_2.13:5.1.2 [S]
[info]   | | +-com.github.ben-manes.caffeine:caffeine:3.0.5
[info]   | | | +-com.google.errorprone:error_prone_annotations:2.10.0
[info]   | | | +-org.checkerframework:checker-qual:3.19.0
[info]   | | |
[info]   | | +-org.scala-lang.modules:scala-java8-compat_2.13:1.0.2 [S]
[info]   | |
[info]   | +-org.typelevel:cats-core_2.13:2.7.0 [S]
[info]   | | +-org.typelevel:cats-kernel_2.13:2.7.0 [S]
[info]   | | +-org.typelevel:simulacrum-scalafix-annotations_2.13:0.5.4 [S]
[info]   | |
[info]   | +-org.typelevel:cats-effect_2.13:3.3.5 [S]
[info]   |   +-org.typelevel:cats-effect-kernel_2.13:3.3.5 [S]
[info]   |   | +-org.typelevel:cats-core_2.13:2.7.0 [S]
[info]   |   |   +-org.typelevel:cats-kernel_2.13:2.7.0 [S]
[info]   |   |   +-org.typelevel:simulacrum-scalafix-annotations_2.13:0.5.4 [S]
[info]   |   |
[info]   |   +-org.typelevel:cats-effect-std_2.13:3.3.5 [S]
[info]   |     +-org.typelevel:cats-effect-kernel_2.13:3.3.5 [S]
[info]   |       +-org.typelevel:cats-core_2.13:2.7.0 [S]
[info]   |         +-org.typelevel:cats-kernel_2.13:2.7.0 [S]
[info]   |         +-org.typelevel:simulacrum-scalafix-annotations_2.13:0.5.4 [S]
```

and if I want reverse lookup

```
scala-weather> whatDependsOn org.typelevel cats-effect_2.13
[info] org.typelevel:cats-effect_2.13:3.3.5 [S]
[info]   +-com.snowplowanalytics:scala-lru-map_2.13:0.6.0 [S]
[info]     +-com.snowplowanalytics:scala-weather_2.13:1.0.0+9-370dff64+20220302-1204-..
```

pondzix commented on project/Dependencies.scala:
```diff
@@ -21,10 +21,9 @@ object Dependencies {
+    val scalaj = "2.4.1"
 
     // Scala (test only)
     val specs2 = "4.4.1"
```
bump specs -> 4.13.2?

pondzix commented on CHANGELOG:
```diff
@@ -1,3 +1,12 @@
+Version 1.0.1 (2022-02-28)
```
I'm wondering whether it should be 2.0.0 as it's we do backward incompatible dependencies upgrades?

oguzhanunlu replied:
yeah

oguzhanunlu commented:
I addressed your feedback and had to open a new PR at https://github.com/snowplow/scala-forex/pull/189 to reflect release version in branch name, hence closing this PR
